File: app/src/main/java/com/example/notes/NotesDataBase.java
package com.example.notes;

import android.content.Context;

import androidx.room.Database;
import androidx.room.Room;
import androidx.room.RoomDatabase;

@Database(entities = Note.class, version = 1)
abstract class NotesDataBase extends RoomDatabase {

    private static NotesDataBase instance;
    public abstract NoteDao noteDao();

    public static synchronized NotesDataBase getInstance(Context context) {
        if (instance == null) {
            instance = Room.databaseBuilder(context.getApplicationContext(),
                    NotesDataBase.class, "notes_db")
                    .fallbackToDestructiveMigration()
                    .build();
        }
        return instance;
    }
}

File: app/src/main/java/com/example/notes/NotesDataBase.kt
package com.example.notes

import android.content.Context

import androidx.room.Database
import androidx.room.Room
import androidx.room.RoomDatabase

@Database(entities = [Note::class], version = 1)
internal abstract class NotesDataBase : RoomDatabase() {
    abstract fun noteDao(): NoteDao

    companion object {

        private var instance: NotesDataBase? = null

        @Synchronized
        fun getInstance(context: Context): NotesDataBase {
            if (instance == null) {
                instance = Room.databaseBuilder(
                    context.applicationContext,
                    NotesDataBase::class.java, "notes_db"
                )
                    .fallbackToDestructiveMigration()
                    .build()
            }
            return instance!!
        }
    }
}
